README.md

```diff
@@ -120,6 +120,36 @@ jobs:
           curl -u ${{ secrets.ARTIFACTS_USER }}:${{ secrets.ARTIFACTS_PASSWORD }} ${{ steps.artifacts-get.outputs.link }}/file1 -o file1
 ```
 
+### Index (beta)
+
+Artifacts index by default the following key:
+* commit
+* shortcommit
+* sha
+* branch
+
+Giving you the ability to list your build Artifacts by branch commits and so on.
+
+Index allow you to set your own key value pairs enabling searching capability with
+parameters that are not indexed by default, and are specific to your repository.
+
+This method is currently in beta, the only key allowed for now is `version`.
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Artifacts Index
+        uses: ./
+        with:
+          url: my.artifacts.url
+          user: ${{ secrets.ARTIFACTS_USER }}
+          password: ${{ secrets.ARTIFACTS_PASSWORD }}
+          key: version
+          value: 1.0.0
```

Comment:
add `method: index` 
and key value as a dict ?

Reply:
As github only supports strings, because those vars are passed through environment variable, I wouldn't go for a dict just for the sake of consistency and applying the same logic that is used on other actions. See per example on the Actions upload artifacts https://github.com/actions/upload-artifact#upload-using-multiple-paths-and-exclusions